core/src/main/java/edu/wpi/grip/core/sources/CameraSource.java

```diff
@@ -37,6 +38,19 @@
      * default path allows the Axis M1011 cameras used in FRC to work when only an IP address is supplied.
      */
     public final static String DEFAULT_IP_CAMERA_PATH = "/mjpg/video.mjpg";
+    private static final int
+            /**
+             * Connecting to a device can take the most time.
+             * This should have a little bit of leeway
+             */
+            IP_CAMERA_CONNECTION_TIMEOUT = 14;
```

Comment:
> The connection timeout should be unnecessary since we just retry the same connection when the timeout is up. If there's a problem that can't be solved within the timeout (like the hostname lookup failed), it would fail immediately anyways.

If we don't have the connection timeout it doesn't give up immediately. It usually hangs for a long time instead. If you try to kill the service while it's trying to connect you cant.


> Also, why aren't these defined in `IPCameraFrameGrabber`?

See the related issue that I opened bytedeco/javacv#322
Because everything relating to the input to the IPCameraFrameGrabber should be in the Camera, not in the logic that runs the camera.